### PR TITLE
Add cmocka

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/cmocka/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  home: https://cmocka.org/
+  license: Apache-2.0
+  license_family: Apache
+  license_file: COPYING
+  summary: An elegant unit testing framework for C with support for mock objects.
+
+  description: |
+    cmocka is ...
+
+    an elegant unit testing framework for C with support for mock objects. It
```
Comment (scopatz) [inline recipes/cmocka/meta.yaml:49]: maybe remove the .... and the newline :wink:

Comment (hmaarrfk) [reply recipes/cmocka/meta.yaml:49]: That's taken from their website. But ok

File: recipes/cmocka/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  url: https://cmocka.org/files/{{ version_major_minor }}/{{ name }}-{{ version }}.tar.xz
+  sha256: f0ccd8242d55e2fd74b16ba518359151f6f8383ff8aef4976e48393f77bba8b6
+
+build:
+  number: 0
+  run_exports:
+    # No correlation between library name and
+    # so names
+    # https://gitlab.com/cmocka/cmocka/-/issues/43
+    - {{ pin_subpackage(name, max_pin='x.x.x') }}
```
Comment (hmaarrfk) [inline recipes/cmocka/meta.yaml:20]: Do i need to create a package `libcmocka` with the `so_version` so I can pin to that correctly?

Seems tedious to have to update that number myself everytime.

Comment (scopatz) [reply recipes/cmocka/meta.yaml:20]: No, no need

Comment (hmaarrfk) [reply recipes/cmocka/meta.yaml:20]: alright, then i'm good with this as is.

**Comment by hmaarrfk** [comment]
Thanks

**Pull request merged**